pkg/mirror-producer/producer_test.go

```diff
@@ -0,0 +1,888 @@
+	Expect(err).NotTo(HaveOccurred())
+	Expect(len(resultMirror.Interfaces)).To(Equal(2))
+
+	// This plugin must return the same interfaces of the previous one in the chain (ovs-cni plugin),
+	// because it doesn't modify interfaces, but it only updates ovsdb.
+	By("Checking that result interfaces are equal to those returned by ovs-cni plugin")
+	hostIface := resultMirror.Interfaces[0]
+	contIface := resultMirror.Interfaces[1]
+	Expect(resultMirror.Interfaces[0]).To(Equal(hostIface))
+	Expect(resultMirror.Interfaces[1]).To(Equal(contIface))
+
+	By("Getting port uuid for the hostIface")
+	portUUID, err := getPortUUIDByName(hostIface.Name)
+	Expect(err).NotTo(HaveOccurred())
+	return portUUID
+}
+
+// function that extracts ports from results and check if every mirror contains those port UUIDs.
+// Since it's not possibile to have mirrors without both ingress and egress,
+// it's enough finding the port in either ingress or egress.
+func checkPortsInMirrors(mirrors []types.Mirror, results ...cnitypes.Result) bool {
+	// build an empty slice of port UUIDs
+	var portUUIDs = make([]string, 0)
+	for _, r := range results {
+		portUUID := getPortUUIDFromResult(r)
+		portUUIDs = append(portUUIDs, portUUID)
+	}
+
+	for _, mirror := range mirrors {
+		By(fmt.Sprintf("Checking that mirror %s is in ovsdb", mirror.Name))
+		mirrorNameOvsdb, err := getMirrorAttribute(mirror.Name, "name")
+		Expect(err).NotTo(HaveOccurred())
+		Expect(mirrorNameOvsdb).To(Equal(mirror.Name))
+
+		if mirror.Ingress {
+			By(fmt.Sprintf("Checking that mirror %s has all ports created by ovs-cni plugin in select_src_port column", mirror.Name))
+			srcPorts, err := getMirrorSrcPorts(mirror.Name)
+			Expect(err).NotTo(HaveOccurred())
+			for _, portUUID := range portUUIDs {
+				Expect(srcPorts).To(ContainElement(portUUID))
+			}
+		}
+
+		if mirror.Egress {
+			By(fmt.Sprintf("Checking that mirror %s has all ports created by ovs-cni plugin in select_dst_port column", mirror.Name))
+			dstPorts, err := getMirrorDstPorts(mirror.Name)
+			Expect(err).NotTo(HaveOccurred())
+			for _, portUUID := range portUUIDs {
+				Expect(dstPorts).To(ContainElement(portUUID))
+			}
+		}
+	}
+	return true
+}
+
+// function to check if a mirror exists by its name
+func isMirrorExists(name string) (bool, error) {
+	output, err := exec.Command("ovs-vsctl", "find", "Mirror", "name="+name).CombinedOutput()
+	if err != nil {
+		return false, fmt.Errorf("failed to check if mirror exists: %v", string(output[:]))
+	}
+	return len(output) > 0, nil
+}
+
+// function to get a portUUID by its name
+func getPortUUIDByName(name string) (string, error) {
+	output, err := exec.Command("ovs-vsctl", "get", "Port", name, "_uuid").CombinedOutput()
+	if err != nil {
+		return "", fmt.Errorf("failed to get port uuid by name: %v", string(output[:]))
+	}
+
+	return strings.TrimSpace(string(output[:])), nil
+}
+
+// function to get a mirror attribute
+func getMirrorAttribute(mirrorName, attributeName string) (string, error) {
```

Comment:
I see that many of these helper functions are duplicated in consumer_test. Please create a test helper package for mirror plugins

Reply:
I created `/pkg/testhelpers/mirror.go` with package `testhelpers`.
Do you prefer this location or inside `/tests/`?

Reply:
It's fine under pkg, at least it does not mix integration and e2e tests